backend/api/Controllers/RobotController.cs

```diff
@@ -258,23 +266,34 @@ public async Task<ActionResult<IsarStopMissionResponse>> StopMission([FromRoute]
             _logger.LogWarning("Could not find robot with id={id}", robotId);
             return NotFound();
         }
-        var response = await _isarService.StopMission(robot);
-        if (!response.IsSuccessStatusCode || response.Content is null)
+        try
         {
-            _logger.LogError("Could not stop mission on robot: {robotId}", robotId);
+            var response = await _isarService.StopMission(robot);
+            if (!response.IsSuccessStatusCode || response.Content is null)
+            {
+                _logger.LogError("Could not stop mission on robot: {robotId}", robotId);
 
-            int statusCode = (int)response.StatusCode;
+                int statusCode = (int)response.StatusCode;
 
-            // If error is caused by user (400 codes), let them know
-            if (statusCode is >= 400 and < 500)
-                return new StatusCodeResult(statusCode);
+                // If error is caused by user (400 codes), let them know
+                if (statusCode is >= 400 and < 500)
+                    return new StatusCodeResult(statusCode);
 
+                return new StatusCodeResult(StatusCodes.Status502BadGateway);
+            }
+
+            string? responseContent = await response.Content.ReadAsStringAsync();
+            var isarResponse = JsonSerializer.Deserialize<IsarStopMissionResponse>(responseContent);
+            return Ok(isarResponse);
+        }
+        catch (HttpRequestException e)
```

Comment:
Reduce the try-clause to only contain the call that is being tested

Reply:
Done.